awx/main/models/receptor_address.py

```diff
@@ -0,0 +1,65 @@
+from django.db import models
+from django.core.validators import MinValueValidator, MaxValueValidator
+from django.utils.translation import gettext_lazy as _
+from awx.api.versioning import reverse
+
+
+class ReceptorAddress(models.Model):
+    class Meta:
+        app_label = 'main'
+        constraints = [
+            models.UniqueConstraint(
+                fields=["address", "protocol"],
+                name="unique_receptor_address",
+                violation_error_message=_("Receptor address + protocol must be unique."),
+            )
+        ]
+
+    class Protocols(models.TextChoices):
```

Comment:
pull out of ReceptorAddress and just call it ReceptorProtocols so that it can easily be used elsewhere